File: Code/HRTest/HRTest/ViewController.m
//
//  ViewController.m
//  HRTest
//
//  Created by Jake Day on 2/26/19.
//  Copyright © 2019 Jake Day. All rights reserved.
//

#import <ResearchKit/ResearchKit.h>
#import "ViewController.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Create a step
    ORKInstructionStep *myStep = [[ORKInstructionStep alloc] initWithIdentifier:@"intro"];
    myStep.title = @"Welcome to ResearchKit";
    
    // Create a task
    ORKOrderedTask *task = [[ORKOrderedTask alloc] initWithIdentifier:@"task" steps:@[myStep]];
    
    // Present the task
    ORKTaskViewController *taskViewController = [[ORKTaskViewController alloc] initWithTask:task taskRunUUID:nil];
    taskViewController.delegate = self;
    [self presentViewController:taskViewController animated:YES completion:nil];
}

- (void)taskViewController:(ORKTaskViewController *)taskViewController didFinishWithReason:(ORKTaskViewControllerFinishReason)reason error:(nullable NSError *)error {
    ORKTaskResult *taskResult = [taskViewController result];
    
    [self dismissViewControllerAnimated:YES completion:nil];
}


@end

File: Code/HRTestSwift/HRTestSwift/ViewController.swift
//
//  ViewController.swift
//  HRTestSwift
//
//  Created by Jake Day on 3/16/19.
//  Copyright © 2019 Jake Day. All rights reserved.
//

import UIKit
import ResearchKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
    }
    
    override func viewDidAppear(_ animated: Bool) {
        // Create a step
        let myStep = ORKHrCaptureStep(identifier: "hr")
        myStep.title = "Heart Rate Test"
        // Create a task
        let task = ORKOrderedTask(identifier: "task", steps: [myStep])
        // Present the task
        let taskViewController = ORKTaskViewController(task: task, taskRun: nil)
        taskViewController.delegate = self as? ORKTaskViewControllerDelegate
        present(taskViewController, animated: true, completion: nil)
    }
    
    func taskViewController(_ taskViewController: ORKTaskViewController, didFinishWith reason: ORKTaskViewControllerFinishReason, error: Error?) {
        let taskResult = taskViewController.result
        // do something with result
        
        // dismiss task view controller
        dismiss(animated: true, completion: nil)
    }


}

